src/library/tactic/simp_lemmas.cpp

```diff
@@ -1227,7 +1227,7 @@ static bool instantiate_emetas(type_context & ctx, list<expr> const & _emetas, l
     lean_assert(emetas.size() == instances.size());
     for (unsigned i = 0; i < emetas.size(); ++i) {
         expr m = emetas[i];
-        unsigned mvar_idx = emetas.size() - 1 - i;
+        unsigned mvar_idx = to_meta_idx(m);
         expr m_type = ctx.instantiate_mvars(ctx.infer(m));
         // TODO(Leo, Daniel): do we need the following assertion?
         // lean_assert(!has_expr_metavar(m_type));
```

Comment:
@leodemoura Note, there is quite some duplication between this `instantiate_emetas` that is used in `refl_lemma_rewrite` and the "proper" one below.

Reply:
I am aware of the duplication. It will be addressed when I get back to the simplifier. I had to stop the cleanup to work on the fresh name and cache problems.